tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/TestUtil.java

```diff
@@ -27,46 +21,54 @@ private TestUtil() {
     }
 
     /**
-     * Compares a collection of results with the collection of the gold standard and returns an
-     * {@link ExplicitEvaluationResults}.
+     * compares the results with the expected results and creates a new {@link EvaluationResults}.
      * 
-     * @param results      Collection of Strings representing the results
-     * @param goldStandard Collection of Strings representing the gold standard
+     * @param arDoCoResult the ArDoCoResult
+     * @param results      Collection representing the results
+     * @param goldStandard Collection representing the gold standard
+     * @param tlr          defines how to calculate the true negatives (whether for tlr or not)
      * @return the result of the comparison
      */
-    public static ExplicitEvaluationResults<String> compare(Collection<String> results, Collection<String> goldStandard) {
-        Set<String> distinctTraceLinks = new HashSet<>(results);
-        Set<String> distinctGoldStandard = new HashSet<>(goldStandard);
+    public static <T> EvaluationResults<T> compare(ArDoCoResult arDoCoResult, Collection<T> results, Collection<T> goldStandard, boolean tlr) {
```

Comment:
Instead of `boolean tlr`, I suggest to add a class that encapsulates the function extraction of `trueNegatives` to ensure that extensions are possible in the future.